snmp/assets/configuration/spec.yaml

```diff
@@ -139,6 +139,63 @@ files:
         value:
           type: boolean
           example: true
+      - name: ping
+        description: |
+          Configure ICMP pings for all hosts in SNMP autodiscovery
+          Each device will be pinged with these settings each time the SNMP
+          check is run.
+
+          By default, we try to use an unprivileged UDP socket to send ICMP
+          pings, but some linux systems require using a raw socket.
+          See details at the link below:
+          https://sturmflut.github.io/linux/ubuntu/2015/01/17/unprivileged-icmp-sockets-on-linux/
+          
+          If `linux.use_raw_socket` is set, you MUST enable the ping module
+          of the system-probe for elevated privileges. See
+          system-probe.yaml.example for details.
+          
+          Example:
+          ping:
+            enabled: true             # disabled by default
+            timeout: 3000             # timeout in milliseconds
+            count: 2                  # number of ping packets to send per check run
+            interval: 20              # time between sending pings (up to `count` packets) in milliseconds
+            linux:                    # linux specific configuration
+              use_raw_socket: true    # send pings in a privileged fashion using a raw socket
+        options:
+        - name: enabled
+          description: Enables/Disables pinging devices
+          value:
+            type: boolean
+            example: true
+        - name: count
+          description: (optional) Number of pings to send per check run
```

Comment:
Should we specify the default for this and other optional parameters?

Reply:
Great idea

Reply:
Turns out the generated file adds formats it and adds the default value to it